lua/hbac/init.lua

```diff
@@ -82,17 +82,58 @@ M.setup = function(opts)
 end
 
 
-local function close_unused()
+M.close_unpinned = function()
 	local curbufnr = vim.api.nvim_get_current_buf()
 	local buflist = vim.api.nvim_list_bufs()
 	for _, bufnr in ipairs(buflist) do
-		if vim.bo[bufnr].buflisted and bufnr ~= curbufnr and not M.persistant_buffers[bufnr] then
+		if vim.bo[bufnr].buflisted and bufnr ~= curbufnr and not M.pinned_buffers[bufnr] then
 			vim.cmd('bd ' .. tostring(bufnr))
 		end
+		print("Hbac: Closed unpinned buffers")
 	end
 end
 
+
+M.pin = function(bufnr)
+  M.pin_buffer(bufnr)
+  print("Hbac: Buffer pinned")
+end
+
+M.unpin = function(bufnr)
+	bufnr = bufnr or vim.api.nvim_get_current_buf()
+	M.pinned_buffers[bufnr] = nil
+  print("Hbac: Buffer unpinned")
+end
```

Comment:
Since this is effectively a toggle of `pin_buffer`, we could refactor `pin_buffer` to also accept a second arg, maybe rename it to `toggle_pin`, and that might also resolve the confusion between `pin` vs. `pin`buffer` mentioned above. I wanted to hold off before a bigger refactor like that.

Reply:
Right, this could be just one function toggle_pin. I didn't get why would it need a second arg. It should only need the bufnr I think.